lib/ansible/modules/windows/win_domain_computer.ps1

```diff
@@ -0,0 +1,183 @@
+  If (-not $sam_account_name.EndsWith("$")) {
+    Fail-Json $result "sam_account_name must end in $"
+  }
+  $enabled = Get-AnsibleParam -obj $params -name "enabled" -default $true | ConvertTo-Bool
+  $description = Get-AnsibleParam -obj $params -name "description" -default "" -resultobj $result
+  $state =(Get-AnsibleParam -obj $params -name "state" -ValidateSet "Present","Absent" -default "Present").ToLower()
+  If ($state -eq "present") {
+    $dns_hostname = Get-AnsibleParam -obj $params -name "dns_hostname" -failifempty $true -resultobj $result
+    $ou = Get-AnsibleParam -obj $params -name "ou" -failifempty $true -resultobj $result
+    $distinguished_name = "CN=$name,$ou"
+
+    $desired_state = @{
+      name = $name
+      sam_account_name = $sam_account_name
+      dns_hostname = $dns_hostname
+      ou = $ou
+      distinguished_name = $distinguished_name
+      description = $description
+      enabled = $enabled
+      state = $state
+    }
+  } Else {
+    $desired_state = @{
+      name = $name
+      state = $state
+    }
+  }
+
+  Return $desired_state
+}
+# ------------------------------------------------------------------------------
+Function Get-InitialState($desired_state) {
+  # Test computer exists
+  $computer = Try { Get-ADComputer `
+      -Identity $desired_state.name `
+      -Properties DistinguishedName,DNSHostName,Enabled,Name,SamAccountName,Description,ObjectClass
+    } Catch { $null }
+  If ($computer) {
+      $initial_state = @{
+      name = $computer.Name
+      sam_account_name = $computer.SamAccountName
+      dns_hostname = $computer.DNSHostName
+      # Get OU from regexp that removes all characters to the first ","
+      ou = $computer.DistinguishedName -creplace "^[^,]*,",""
+      distinguished_name = $computer.DistinguishedName
+      description = $computer.Description
+      enabled = $computer.Enabled
+      state = "present"
+    }
+  } Else {
+    $initial_state = @{
+      name = $desired_state.name
+      state = "absent"
+    }
+  }
+
+  return $initial_state
+}
+
+# ------------------------------------------------------------------------------
+Function Set-ConstructedState($initial_state, $desired_state) {
+  Try {
+    Set-ADComputer `
+      -Identity $desired_state.name `
+      -SamAccountName $desired_state.name `
+      -DNSHostName $desired_state.dns_hostname `
+      -Enabled $desired_state.enabled `
+      -Description $desired_state.description `
+      -WhatIf:$check_mode
+  } Catch { Fail-Json $result "Error in function Set-ConstructedState" }
+
+  If ($initial_state.distinguished_name -ne $desired_state.distinguished_name)
+  {
+    # Move computer to OU
+    Try {
+      Get-ADComputer $desired_state.name |
+          Move-ADObject `
+            -TargetPath $desired_state.ou `
+            -Confirm:$False `
+            -WhatIf:$check_mode
+    } Catch { Fail-Json $result "Error in function Move-ADObject" }
+  }
+  Set-Attr $result "changed" $true
```

Comment:
Just do `$result.changed = $true`, no need for `Set-Attr` now result is a hashtable.